renderer/backend/DrawBatchExecutor.h

```diff
@@ -0,0 +1,55 @@
+/*****************************************************************************
+                    The Dark Mod GPL Source Code
+
+ This file is part of the The Dark Mod Source Code, originally based
+ on the Doom 3 GPL Source Code as published in 2011.
+
+ The Dark Mod Source Code is free software: you can redistribute it
+ and/or modify it under the terms of the GNU General Public License as
+ published by the Free Software Foundation, either version 3 of the License,
+ or (at your option) any later version. For details, see LICENSE.TXT.
+
+ Project: The Dark Mod (http://www.thedarkmod.com/)
+
+******************************************************************************/
+#pragma once
+#include "UniversalGpuBuffer.h"
+#include "../tr_local.h"
+
+struct DrawElementsIndirectCommand {
+    uint count;
+    uint instanceCount;
+    uint firstIndex;
+    uint baseVertex;
+    uint baseInstance;
+};
+
+class DrawBatchExecutor {
+public:
+	void Init();
+	void Destroy();
+
+	void BeginBatch(int maxDrawCalls);
+	void AddDrawVertSurf(const drawSurf_t *surf);
+	void AddShadowSurf( const drawSurf_t * surf );
+	void DrawBatch();
+	void Lock();
```

Comment:
It is not clear what this method is for.
A comment would be welcome.

Logically, `DrawBatch` can `Lock` internally at the end of the function...

Reply:
Each `Lock` issues a glFenceSync, which are expensive. Yes, you could do it after every DrawBatch, but performance would suffer, so I do it explicitly only at the end of a frame (or implicitly when the buffer wraps around). I can add some comments.

Reply:
Add comment for the Lock method at least, please.